File: leetcode/Easy/724-find-pivot-index/solution.cpp
class Solution {
public:
      int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        if(n==0) return -1;
        int count =0;
        for(int i=0;i<n;i++){
            count += nums[i];
        }
        int sum =0;
        for(int i=0;i<n;i++){
            count -= nums[i];
            if(count == sum){
                return i;
            }
             sum += nums[i];
        }
        return -1;

    }
};

File: leetcode/Easy/724-find-pivot-index/solution.java
class Solution {
    public int pivotIndex(int[] nums) {
        int total_sum = 0;
        for(int i : nums){
            total_sum += i;
        }
        int sum = 0;
        for(int i = 0; i<nums.length;i++){
            total_sum -= nums[i];
            sum += nums[i];

            if(total_sum == sum - nums[i]){
                return i;
            }
        }
        return -1;
    }
}
